### PR TITLE
[AIR] [Data] Add case for `Dict[str, np.array]` batches in `DummyTrainer` read bytes calculation

### PR DESCRIPTION
**Pull request by scottjlee**
## Why are these changes needed?
When following the `DummyTrainer` example in [our docs](https://docs.ray.io/en/latest/ray-air/check-ingest.html#debugging-ingest-with-the-dummytrainer), the `DummyTrainer` does not correctly calculate the bytes read in the case of `Dict[str, np.array]` batches. This PR ensures that we use the `numpy` batch format to get this expected batch type, then iterates over each array in the dict and sums up `arr.nbytes` to calculate the total size.

## Related issue number

Related to #36686

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottjlee** [comment]
While looking at this issue, we discovered another underlying bug where the `DataIterator.iter_batches()` returned a batch of type `pandas`, when we are expecting `numpy` (from strict mode). Do we need to add a `batch_format` conversion to the beginning of `DataIterator.iter_batches()` [here](https://github.com/ray-project/ray/blob/master/python/ray/data/iterator.py#L133), as we have in similar `Dataset.iter_batches()` [method](https://github.com/ray-project/ray/blob/master/python/ray/data/dataset.py#L2953), map_batches, etc?

**Comment by scottjlee** [comment]
CI failures look unrelated to me and/or are present on master

File: python/ray/air/util/check_ingest.py [hunk 1]
```diff
@@ -93,6 +93,9 @@ def train_loop_per_worker():
                         )
                     elif isinstance(batch, np.ndarray):
                         bytes_read += batch.nbytes
+                    elif isinstance(batch, dict):
```
Comment (justinvyu) [inline python/ray/air/util/check_ingest.py:96]: It's only possible for this to be a dict of numpy arrays? Any other possible formats?

**Review by justinvyu — APPROVED** [review]
Thanks. Is the numpy format thing no longer an issue? Will this still not work for reading a parquet format, which defaults to pandas?

File: python/ray/air/util/check_ingest.py (hunk 1 above)
Comment (scottjlee) [reply python/ray/air/util/check_ingest.py:96]: With strict mode enabled, this should be defaulting to `numpy` as the default batch format. We missed a case with the `DataIterator` where we did not enforce this default batch format, which we are fixing in a separate PR: https://github.com/ray-project/ray/pull/36686

So, once the above PR is merged, the numpy format will no longer be an issue, as the default batch format will now be numpy.

**Pull request merged**